[WPE] WPE Platform: add support for touch events in DRM

carlosgcampos opened the pull request:
#### eeca106abd6024f3aa93f7bf53a5edbf0b7d96a7
<pre>
[WPE] WPE Platform: add support for touch events in DRM
<a href="https://bugs.webkit.org/show_bug.cgi?id=282230">https://bugs.webkit.org/show_bug.cgi?id=282230</a>

Reviewed by Alejandro G. Castro and Adrian Perez de Castro.

* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.cpp:
(WPE::DRM::Seat::processEvent):
(WPE::DRM::Seat::handleTouchDownEvent):
(WPE::DRM::Seat::handleTouchUpEvent):
(WPE::DRM::Seat::handleTouchMotionEvent):
(WPE::DRM::libinput_event_touch::for):
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.h:

Canonical link: <a href="https://commits.webkit.org/286075@main">https://commits.webkit.org/286075@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/32d1e71206c7875081503f5e31de1b8e8a81aded

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73946 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53375 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26757 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78296 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25184 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62508 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1160 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58147 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16494 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77013 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48288 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63611 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38547 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45120 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23517 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66657 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21449 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79838 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1263 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/663 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66462 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1406 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63625 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65743 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9636 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7816 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11535 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1227 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/4061 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1256 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1244 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1263 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM!

aperezdc commented on Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.cpp:
```diff
@@ -182,6 +182,18 @@ void Seat::processEvent(struct libinput_event* event)
     case LIBINPUT_EVENT_KEYBOARD_KEY:
         handleKeyEvent(libinput_event_get_keyboard_event(event));
         break;
+    case LIBINPUT_EVENT_TOUCH_DOWN:
+        handleTouchDownEvent(libinput_event_get_touch_event(event));
+        break;
+    case LIBINPUT_EVENT_TOUCH_UP:
+        handleTouchUpEvent(libinput_event_get_touch_event(event));
+        break;
+    case LIBINPUT_EVENT_TOUCH_MOTION:
+        handleTouchMotionEvent(libinput_event_get_touch_event(event));
+        break;
+    case LIBINPUT_EVENT_TOUCH_CANCEL:
+        handleTouchCancelEvent(libinput_event_get_touch_event(event));
+        break;
```
I am surprised that this does not handle `LIBINPUT_EVENT_TOUCH_FRAME`: in order to support multitouch input, multiple events may be received at the same time, and the “frame” marker indicates when all the simultaneous multitouch events have been picked. 

Or are you leaving multitouch support for a follow-up patch?

Also, since libinput 1.19 pinch, swipe, and hold gestures (`LIBINPUT_EVENT_GESTURE_*`) are directly recognized by libinput, we may want to use those e.g. touch-hold to trigger the context menu.

carlosgcampos committed:
[WPE] WPE Platform: add support for touch events in DRM
https://bugs.webkit.org/show_bug.cgi?id=282230

Reviewed by Alejandro G. Castro and Adrian Perez de Castro.

* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.cpp:
(WPE::DRM::Seat::processEvent):
(WPE::DRM::Seat::handleTouchDownEvent):
(WPE::DRM::Seat::handleTouchUpEvent):
(WPE::DRM::Seat::handleTouchMotionEvent):
(WPE::DRM::libinput_event_touch::for):
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.h:

Canonical link: https://commits.webkit.org/286075@main
